lib/active_merchant/billing/gateways/payflow.rb

```diff
@@ -188,6 +189,53 @@ def build_credit_card_request(action, money, credit_card, options)
           end
         end
         xml.target!
```

Comment:
So I'm not super familiar with `Builder`, but it looks like what `target!` does is return the built XML as a string so far? Since this call to `target!` is no longer the return value, unless there is some necessary side-effect from calling it before here before the final expression, the value returned from `xml.target!` would just be discarded. Am I reading things right?

Reply:
That is indeed what Builder is doing here. The secret side effect is that Nokogiri, which we're about to use to disassemble and reassemble the XML, doesn't know what to do with a Builder XML object. I didn't feel super enthusiastic about converting the whole method to Nokogiri given time constraints, so turning it into a string and passing it along seemed simplest.